ndarray/src/main/java/org/tensorflow/ndarray/index/Slice.java

```diff
@@ -0,0 +1,111 @@
+/*
+  Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================
+ */
+package org.tensorflow.ndarray.index;
+
+import java.util.StringJoiner;
+import org.tensorflow.ndarray.impl.dimension.Dimension;
+
+final class Slice implements TensorIndex {
```

Comment:
If we take the `sliceTo`/`sliceFrom` route, having distinct implementation of `Index` (i.e. `SliceTo` and `SliceFrom`) could reduce the complexity/cost of each case and avoid the boxed type conversion. There is not much gain having them all implemented by a single class.